commands/command_filter.go

```diff
@@ -20,6 +20,37 @@ const (
 	SmudgeOperation
 )
 
+type InputFileHdr struct {
+	FileName string
+	FileLen  uint32
+}
+
+func (h *InputFileHdr) Read(r io.Reader) error {
+	// Read file name length
+	var fileNameLen uint32
+	if err := binary.Read(r, binary.LittleEndian, &fileNameLen); err != nil {
+		return errutil.Errorf(err, "Error reading filename length.")
+	}
+
+	// Read file name
+	h.FileName = ""
```

Comment:
Is there a reason you have to explicitly empty this out? Go's zero-value initialization will make this string be `""` by default.